### ISSUE TITLE
Heap corruption / SIGABRT when compiling `ConvTranspose2d` + normalization layers with `torch.compile(backend='inductor')`

### ISSUE THREAD

**Issue by jasminetrail** [pytorch/pytorch#181688]
### 🐛 Describe the bug

Running `torch.compile(model, backend='inductor')` on a model that contains `nn.ConvTranspose2d` followed by a normalization layer (`nn.GroupNorm`, `nn.InstanceNorm2d`, or `nn.BatchNorm2d`) causes the process to abort with a glibc heap corruption error or a silent `SIGABRT`.

The crash occurs during the compiled forward pass and is not reproducible under eager execution.  It is a **memory safety** issue in Inductor's generated C++ kernel — the process dies inside the C malloc allocator, indicating a write past the end of an allocated buffer or a use-after-free in the compiled code.

## Minimal Reproducer

```python
import torch
import torch.nn as nn

torch.manual_seed(0)
x = torch.randn([9, 8, 14, 11])

conv_t = nn.ConvTranspose2d(8, 6, 2)
gn     = nn.GroupNorm(6, 6).eval()
ln1    = nn.LayerNorm([12])
ln2    = nn.LayerNorm([12])

def model():
    t = torch.clamp(x, min=-2.0, max=2.0)
    t = conv_t(t)
    t = gn(t)
    t = ln1(t)
    t = ln2(t)
    return t

# Eager forward works fine
out = model()
assert torch.isfinite(out).all()

# Compiled forward crashes the process
compiled = torch.compile(model, backend='inductor')
compiled()   # ← SIGABRT / heap corruption here
```

### Error logs

Most programs crash **silently** (bare `SIGABRT`, no output to stderr).  When the allocator does emit a message, typical examples are:

```
malloc_consolidate(): invalid chunk size
Aborted (core dumped)
```

```
corrupted size vs. prev_size
Aborted (core dumped)
```

```
double free or corruption (out)
Aborted (core dumped)
```

The exact message varies between runs and between programs; the underlying cause in all cases is heap metadata corruption in the Inductor-generated kernel.

### Versions

Collecting environment information...
PyTorch version: 2.11.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.3) 11.4.0
Clang version: 15.0.0 ([git@github.com](mailto:git@github.com):llvm/llvm-project.git 4ba6a9c9f65bbc8bd06e3652cb20fd4dfc846137)
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Mar 3 2026, 11:56:32) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.8.0-94-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture: x86_64
CPU op-mode(s): 32-bit, 64-bit
Address sizes: 52 bits physical, 57 bits virtual
Byte Order: Little Endian
CPU(s): 384
On-line CPU(s) list: 0-383
Vendor ID: AuthenticAMD
Model name: AMD EPYC 9684X 96-Core Processor
CPU family: 25
Model: 17
Thread(s) per core: 2
Core(s) per socket: 96
Socket(s): 2
Stepping: 2
BogoMIPS: 5099.98
Flags: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin cppc amd_ibpb_ret arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid overflow_recov succor smca fsrm flush_l1d debug_swap ibpb_exit_to_user
Virtualization: AMD-V
L1d cache: 6 MiB (192 instances)
L1i cache: 6 MiB (192 instances)
L2 cache: 192 MiB (192 instances)
L3 cache: 2.3 GiB (24 instances)
NUMA node(s): 2
NUMA node0 CPU(s): 0-95,192-287
NUMA node1 CPU(s): 96-191,288-383
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit: Not affected
Vulnerability L1tf: Not affected
Vulnerability Mds: Not affected
Vulnerability Meltdown: Not affected
Vulnerability Mmio stale data: Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed: Not affected
Vulnerability Spec rstack overflow: Mitigation; Safe RET
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1: Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2: Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds: Not affected
Vulnerability Tsx async abort: Not affected
Vulnerability Vmscape: Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] nvidia-cublas==13.1.0.3
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cudnn-cu13==9.19.0.56
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.0
[pip3] nvidia-nccl-cu13==2.28.9
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvtx==13.0.85
[pip3] torch==2.11.0
[pip3] triton==3.6.0
[conda] Could not collect

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by jasminetrail** [comment]
Here is the [gist](https://colab.research.google.com/gist/jasminetrail/ea635a6877bd1ccc22094ea25cac8620/issue_181688.ipynb)

**Comment by aorenste** [comment]
I tried to reproduce this issue on PyTorch nightly (`2.12.0.dev20260330+cu130`) on NVIDIA PG509-210. The behavior is flaky (2/3 runs failed).

2 out of 3 runs crashed with heap corruption errors matching the reported issue. Run 1 passed. Run 2 segfaulted (SIGSEGV, exit code 139). Run 3 aborted with 'corrupted double-linked list' (SIGABRT, exit code 134). These are the exact memory safety errors described in the issue: heap corruption in the Inductor-generated C++ kernel when compiling ConvTranspose2d + normalization layers.

I attempted bisection across 30 days of nightlies but could not determine a clear transition point.

<details>
<summary>Error output</summary>

```
Eager forward: PASS
Compiled forward: PASS

corrupted double-linked list
/tmp/tmptg_x6w43: line 3: 1342973 Aborted                 (core dumped) python repro.py

ERROR conda.cli.main_run:execute(49): `conda run python repro.py` failed. (See above for error)
```
</details>

<details>
<summary>Repro script I used</summary>

```python
import torch
import torch.nn as nn

torch.manual_seed(0)
x = torch.randn([9, 8, 14, 11])

conv_t = nn.ConvTranspose2d(8, 6, 2)
gn     = nn.GroupNorm(6, 6).eval()
ln1    = nn.LayerNorm([12])
ln2    = nn.LayerNorm([12])

def model():
    t = torch.clamp(x, min=-2.0, max=2.0)
    t = conv_t(t)
    t = gn(t)
    t = ln1(t)
    t = ln2(t)
    return t

# Eager forward works fine
out = model()
assert torch.isfinite(out).all()
print("Eager forward: PASS")

# Compiled forward crashes the process
compiled = torch.compile(model, backend='inductor')
compiled_out = compiled()
print("Compiled forward: PASS")

# Verify outputs match
if torch.allclose(out, compiled_out, atol=1e-5):
    print("Output match: PASS")
else:
    max_diff = (out - compiled_out).abs().max().item()
    print(f"Output match: FAIL (max diff = {max_diff})")

```
</details>

<details>
<summary>My environment</summary>

```
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: CentOS Stream 9 (x86_64)
GCC version: (GCC) 11.5.0 20240719 (Red Hat 11.5.0-14)
Clang version: Could not collect
CMake version: version 4.3.1
Libc version: glibc-2.34

Python version: 3.13.13 | packaged by conda-forge | (main, Apr  8 2026, 02:00:33) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.13.2-0_fbk11_0_g599ea5da5981-x86_64-with-glibc2.34
Is CUDA available: N/A
CUDA runtime version: 12.9.86
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: 
GPU 0: NVIDIA PG509-210
GPU 1: NVIDIA PG509-210
GPU 2: NVIDIA PG509-210
GPU 3: NVIDIA PG509-210
GPU 4: NVIDIA PG509-210
GPU 5: NVIDIA PG509-210
GPU 6: NVIDIA PG509-210
GPU 7: NVIDIA PG509-210

Nvidia driver version: 580.82.07
cuDNN version: Probably one of the following:
/usr/lib64/libcudnn.so.8.8.0
/usr/lib64/libcudnn.so.9.5.0
/usr/lib64/libcudnn_adv.so.9.5.0
/usr/lib64/libcudnn_adv_infer.so.8.8.0
/usr/lib64/libcudnn_adv_train.so.8.8.0
/usr/lib64/libcudnn_cnn.so.9.5.0
/usr/lib64/libcudnn_cnn_infer.so.8.8.0
/usr/lib64/libcudnn_cnn_train.so.8.8.0
/usr/lib64/libcudnn_engines_precompiled.so.9.5.0
/usr/lib64/libcudnn_engines_runtime_compiled.so.9.5.0
/usr/lib64/libcudnn_graph.so.9.5.0
/usr/lib64/libcudnn_heuristic.so.9.5.0
/usr/lib64/libcudnn_ops.so.9.5.0
/usr/lib64/libcudnn_ops_infer.so.8.8.0
/usr/lib64/libcudnn_ops_train.so.8.8.0
Is XPU available: N/A
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A
Caching allocator config: N/A

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        46 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               192
On-line CPU(s) list:                  0-191
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) Platinum 8339HC CPU @ 1.80GHz
CPU family:                           6
Model:                                85
Thread(s) per core:                   2
Core(s) per socket:                   24
Socket(s):                            4
Stepping:                             11
Frequency boost:                      enabled
CPU(s) scaling MHz:                   100%
CPU max MHz:                          1801.0000
CPU min MHz:                          800.0000
BogoMIPS:                             3600.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local avx512_bf16 dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi pku ospke avx512_vnni md_clear flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            3 MiB (96 instances)
L1i cache:                            3 MiB (96 instances)
L2 cache:                             96 MiB (96 instances)
L3 cache:                             132 MiB (4 instances)
NUMA node(s):                         4
NUMA node0 CPU(s):                    0-23,96-119
NUMA node1 CPU(s):                    24-47,120-143
NUMA node2 CPU(s):                    48-71,144-167
NUMA node3 CPU(s):                    72-95,168-191
Vulnerability Gather data sampling:   Vulnerable: No microcode
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Vulnerable
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Vulnerable
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Vulnerable
Vulnerability Spectre v1:             Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:             Vulnerable; IBPB: disabled; STIBP: disabled; PBRSB-eIBRS: Vulnerable; BHI: Vulnerable
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.4.4
[pip3] nvidia-cublas==13.1.1.3
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cudnn-cu13==9.20.0.48
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.1
[pip3] nvidia-nccl-cu13==2.29.7
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvtx==13.0.85
[pip3] torch==2.13.0.dev20260428+cu130
[pip3] torchaudio==2.11.0.dev20260428+cu130
[pip3] torchvision==0.27.0.dev20260428+cu130
[pip3] triton==3.7.0+git88b227e2
[conda] mkl                       2025.3.1            h0e700b2_10    conda-forge
[conda] mkl-include               2025.3.1            hf2ce2f3_10    conda-forge
[conda] nccl                      2.29.3.1             h4d09622_0    conda-forge
[conda] tbb                       2022.3.0             hb700be7_2    conda-forge
```
</details>

**Comment by aorenste** [comment]
Marking as a duplicate of #181618
